src/Aspire.Hosting/OtlpConfigurationExtensions.cs

```diff
@@ -32,6 +32,30 @@ public static void AddOtlpEnvironment(IResource resource, IConfiguration configu
         // Add annotation to mark this resource as having OTLP exporter configured
         resource.Annotations.Add(new OtlpExporterAnnotation());
 
+        RegisterOtlpEnvironment(resource, configuration, environment);
+    }
+
+    /// <summary>
+    /// Configures OpenTelemetry in projects using environment variables.
+    /// </summary>
+    /// <param name="resource">The resource to add annotations to.</param>
+    /// <param name="configuration">The configuration to use for the OTLP exporter endpoint URL.</param>
+    /// <param name="environment">The host environment to check if the application is running in development mode.</param>
+    /// <param name="protocol">The protocol to use for the OTLP exporter. If not set, it will try gRPC then Http.</param>
+    public static void AddOtlpEnvironment(IResource resource, IConfiguration configuration, IHostEnvironment environment, OtlpProtocol protocol)
```

Comment:
The existing `AddOtlpEnvironment` overload should probably call this one to reduce code duplication.